### PR TITLE
numpydoc-ify art3d docstrings

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Convert the art3d docstrings to numpydoc / PEP-8.

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 1]
```diff
@@ -28,22 +28,40 @@
...
+        The direction. Possible values are:
+        - 'x': equivalent to (1, 0, 0)
+        - 'y': euqivalent to (0, 1, 0)
+        - 'z': equivalent to (0, 0, 1)
+        - *None*: euqivalent to (0, 0, 0)
+        - an iterable (x, y, z) is returned unchanged.
+
+    Returns
+    -------
+    np.array((x, y, z))
```
Comment (WeatherGod) [inline lib/mpl_toolkits/mplot3d/art3d.py]: slight quibble, if a tuple was passed in, then a tuple comes back. So array-like, instead?

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 2]
```diff
@@ -59,18 +77,26 @@ def get_dir_vector(zdir):
...
+   """
 
     def __init__(self, x=0, y=0, z=0, text='', zdir='z', **kwargs):
-        '''
-        *x*, *y*, *z*  Position of text
-        *text*         Text string to display
-        *zdir*         Direction of text
-
-        Keyword arguments are passed onto :func:`~matplotlib.text.Text`.
-        '''
```
Comment (WeatherGod) [inline lib/mpl_toolkits/mplot3d/art3d.py]: Question: If a subclass puts a docstring up in the class level (as opposed to the `__init__` level), but the parent class has a docstring in its `__init__`, which one shows up in sphinx? I haven't checked if that is the case here, just wondering.

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 3]
```diff
@@ -526,11 +549,16 @@ def __init__(self, verts, *args, **kwargs):
...
-            string: 'average', 'min' or 'max'
-        '''
+        """
+        Sets the calculation method for the z-order.
+
+        Parameters
+        ----------
+        zsort : bool or {'average', 'min', 'max'}
+            For 'average', 'min', 'max' the z-order is determined by applying
+            the function to the z-orders of the individual polygons. *True* is
```
Comment (WeatherGod) [inline lib/mpl_toolkits/mplot3d/art3d.py]: "the function to the z coordinates of the vertices in the viewer's coordinate system."

Perhaps clearer, more precise. Z-order as a term is a bit loaded in matplotlib.

File: lib/mpl_toolkits/mplot3d/art3d.py (hunk 2 above)
Comment (timhoffm) [reply lib/mpl_toolkits/mplot3d/art3d.py]: I haven't checked either. Just following the [numpydoc guide](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#class-docstring)

>  The constructor (`__init__`) should also be documented here [class docstring], the Parameters section of the docstring details the constructors parameters.

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 4]
```diff
@@ -58,19 +78,27 @@ def get_dir_vector(zdir):
         raise ValueError("'x', 'y', 'z', None or vector of length 3 expected")
 
 
-class Text3D(mtext.Text):
-    '''
-    Text object with 3D position and (in the future) direction.
-    '''
+class Text3D(mtext.Taext):
```
Comment (dstansby) [inline lib/mpl_toolkits/mplot3d/art3d.py]: This should be `mtext.Text`!

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 5]
```diff
@@ -28,22 +28,42 @@
...
+        - 'x': equivalent to (1, 0, 0)
+        - 'y': euqivalent to (0, 1, 0)
+        - 'z': equivalent to (0, 0, 1)
+        - *None*: euqivalent to (0, 0, 0)
+        - an iterable (x, y, z) is returned unchanged.
+
+    Returns
+    -------
+    x, y, z : array-like
+        The direction vector. This is either a numpy.array *zdir* itself if
```
Comment (WeatherGod) [inline lib/mpl_toolkits/mplot3d/art3d.py]: `... numpy.array or *zdir* itself ...`

**Comment by WeatherGod** [comment]
Other than the one little typo, this looks good to me.

**Comment by timhoffm** [comment]
ping @WeatherGod @dstansby All requested changes are applied. Do I have to dismiss the reviews?

**Pull request merged**